### PR TITLE
feat: support tags in flow runs api

### PR DESCRIPTION
**Pull request by abuaboud**
## What does this PR do?

Fixes # (issue)

### CODE REVIEW

File: packages/backend/src/app/database/database-common.ts [hunk 1]
```diff
@@ -10,13 +10,19 @@ export const JSONB_COLUMN_TYPE =
   databaseType === DatabaseType.SQLITE3 ? 'simple-json' : 'jsonb'
 export const BLOB_COLUMN_TYPE =
   databaseType === DatabaseType.SQLITE3 ? 'blob' : 'bytea'
+export const ARRAY_COLUMN_TYPE =
+  databaseType === DatabaseType.SQLITE3 ? 'simple-array' : String
```
Comment (khaledmashaly) [inline packages/backend/src/app/database/database-common.ts:14]: I think `simple-array` works for `postgres` and `sqlite`, no?

File: packages/backend/src/app/database/database-connection.ts [hunk 2]
```diff
@@ -46,3 +47,25 @@ export const databaseConnection =
...
+                    ...params,
+                    [`tag${index}`]: `%${tag}%`,
+                }
+            }, {})
+            return query.andWhere(likeConditions, likeParams)
+        }
+        default:
+            throw new Error(`Unsupported database type: ${databaseType}`)
+    }
+}
```
Comment (khaledmashaly) [inline packages/backend/src/app/database/database-connection.ts:71]: wow! the decision to incorporate `sqlite` is making a lot of noise, we are basically doing everything db-related two times, I think we should either drop support for `sqlite` or if adoption numbers justify the use of `sqlite` let's see if we can switch to another ORM that can make things a little bit uniform across multiple DBs, right now TypeORM does little to nothing in this regard.

File: packages/backend/src/app/database/migration/1692106268660-AddTagsToRun.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+export class AddTagsToRun1692106268660 implements MigrationInterface {
+    name = 'AddTagsToRun1692106268660'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query('ALTER TABLE "flow_run" ADD "tags" text')
+    }
+
+    public async down(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query('ALTER TABLE "flow_run" DROP COLUMN "tags"')
```
Comment (khaledmashaly) [inline packages/backend/src/app/database/migration/1692106268660-AddTagsToRun.ts:11]: I think this migration is not needed, right?

**Review by khaledmashaly — COMMENTED** [review]
Can you please clarify what are tags exactly? Are they just some tags for search purposes? Are they connected to flows? or to connections?

Sorry, I'm kind of confused! Did we discuss this feature?

File: packages/backend/src/app/database/migration/1692106268660-AddTagsToRun.ts (hunk 3 above)
Comment (abuaboud) [reply packages/backend/src/app/database/migration/1692106268660-AddTagsToRun.ts:11]: It's auto generated in case you want to rollback yes.

**Pull request merged**